### PR TITLE
[Android] Input evolution prototype

### PR DESCRIPTION
**Pull request by almedina-ms**
Android input evolution prototype based on its UWP counterpart.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/3872)

### CODE REVIEW

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/inputhandler/validation/ActualValidator.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    @Override
+    public List<View> getViewsForValidation()
+    {
+        return m_inputValidator.getViewsForValidation();
+    }
+
+    /*
+    public View getView()
```
Comment (nesalang) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/inputhandler/validation/ActualValidator.java]: nit: comments

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/inputhandler/validation/TextInputRangeValidator.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package io.adaptivecards.renderer.inputhandler.validation;
+
+import android.widget.EditText;
+
+import java.util.regex.Pattern;
+
+import io.adaptivecards.objectmodel.TextInput;
+
+public abstract class TextInputRangeValidator extends ActualValidator
```
Comment (nesalang) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/inputhandler/validation/TextInputRangeValidator.java]: just curious, are we going to have Range/Regex/Required Validators for other type of input such as Number?

**Comment by shalinijoshi19** [comment]
@paulcam206 what's  the plan here based on your recommendation again? Should we now create a shared feature branch on origin to absorb these changes and move out of @RebeccaAnne's user branch? Or merge directly into master at this point to conserve resources?

**Comment by paulcam206** [comment]
> @paulcam206 what's the plan here based on your recommendation again? Should we now create a shared feature branch on origin to absorb these changes and move out of @RebeccaAnne's user branch? Or merge directly into master at this point to conserve resources?

This PR should be merged as it's currently targeted. Some of the feature branch process hinges on the branch being named `feature/*` (mostly status check requirements and build triggers). Once this branch is merged, we should create a new feature branch at the same point as `rebecch/inputprototype` and push it. Future PRs should be against the feature branch. We can delete `rebecch/inputprototype`, though it might make it a little tricky for anyone that has branches off of it. Not that folks can't get fixed if they target the non-feature branch, but it will require intervention on their part to retarget to the feature branch.

**Comment by shalinijoshi19** [comment]
Closing; We have an updated Android PR

**Pull request closed without merging**